D3521R0.md

```diff
@@ -200,80 +200,46 @@ Given an example like:
 Conceptually: `try_cast<@*type*@>(@*subject*@)` is invoked, which returns an
 optional result that can be tested and dereferenced to match the *subpattern*.
 
-More precisely: Let the result of `try_cast<@*type*@>(@*subject*@)` be `__temp`
-where `try_cast` is looked up by ADL-only (same as `get` for structured bindings).
-The pattern matches if `__temp` contextually converts to `true` and
+More precisely: Let `E` be
+`std::try_cast<std::remove_cv_t<@*type*@>>(@*subject*@)`.  The match succeeds
+if and only if: `E` is well-formed; `E` is implicitly convertible to
+`@*type*@`; `E` contextually converts to `true`; and
 `std::forward_like<decltype(@*subject*@)>(*__temp)` matches *subpattern*.
 
 # Proposed Wording
 
-### [any.synop]{.sref} Header `<any>` synopsis {.unnumbered .unlisted}
+### Insert after `[utility.unreachable]` in header `<utilities>` synopsis ([utility.syn]{.sref}):
 
 ```diff
-  template<class T>
-    T any_cast(const any& operand);
-  template<class T>
-    T any_cast(any& operand);
-  template<class T>
-    T any_cast(any&& operand);
-
-  template<class T>
-    const T* any_cast(const any* operand) noexcept;
-  template<class T>
-    T* any_cast(any* operand) noexcept;
-
-+ template<class T, same_as<any> Any>
-+   const T* try_cast(const Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any&& operand) noexcept;
++ // [utility.try_cast], `try_cast`
++ template<class T>
++   requires same_as<T, remove_cvref_t<T>>
++     constexpr @*unspecified*@ try_cast = @*unspecified*@;
```

Comment:
Did we not upstream `$`?

Reply:
I do not understand this comment at all.

Reply:
@brevzin No, you did not.
@tzlaine In his fork, Barry has an extension where rather than writing `@*unspecified*@`, it's just `$unspecified$`. But he's holding out on us.

Reply:
Ah!  That would be handy.  I really struggle with typing the two characters on either side, when they're far apart and in reverse order on each end.